<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Gibran Hemani" />


<title>Perform fast queries against a massive database of complete GWAS summary data</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Perform fast queries against a massive
database of complete GWAS summary data</h1>
<h4 class="author">Gibran Hemani</h4>


<div id="TOC">
<ul>
<li><a href="#authentication" id="toc-authentication">Authentication</a>
<ul>
<li><a href="#deprecated-google-authentication" id="toc-deprecated-google-authentication">Deprecated Google
authentication</a></li>
<li><a href="#allowance" id="toc-allowance">Allowance</a></li>
</ul></li>
<li><a href="#general-api-queries" id="toc-general-api-queries">General
API queries</a></li>
<li><a href="#get-api-status" id="toc-get-api-status">Get API
status</a></li>
<li><a href="#get-list-of-all-available-studies" id="toc-get-list-of-all-available-studies">Get list of all available
studies</a></li>
<li><a href="#get-list-of-a-specific-study" id="toc-get-list-of-a-specific-study">Get list of a specific
study</a></li>
<li><a href="#extract-particular-associations-from-particular-studies" id="toc-extract-particular-associations-from-particular-studies">Extract
particular associations from particular studies</a></li>
<li><a href="#get-the-tophits-from-a-study" id="toc-get-the-tophits-from-a-study">Get the tophits from a
study</a></li>
<li><a href="#perform-phewas" id="toc-perform-phewas">Perform
PheWAS</a></li>
<li><a href="#ld-clumping" id="toc-ld-clumping">LD clumping</a></li>
<li><a href="#ld-matrix" id="toc-ld-matrix">LD matrix</a></li>
<li><a href="#variant-information" id="toc-variant-information">Variant
information</a></li>
<li><a href="#extracting-gwas-summary-data-based-on-gene-region" id="toc-extracting-gwas-summary-data-based-on-gene-region">Extracting
GWAS summary data based on gene region</a></li>
<li><a href="#genomes-annotations" id="toc-genomes-annotations">1000
genomes annotations</a></li>
</ul>
</div>

<p>The <a href="https://gwas.mrcieu.ac.uk/">OpenGWAS database</a>
comprises over 50,000 curated, QC’d and harmonised complete GWAS summary
datasets and can be queried using an API. See <a href="https://api.opengwas.io/api/">here</a> for documentation on the
API itself. This R package is a wrapper to make generic calls to the
API, plus convenience functions for specific queries.</p>
<div id="authentication" class="section level2">
<h2>Authentication</h2>
<p>From 1st May 2024, most queries to the OpenGWAS API will require user
authentication. For more information on why this is necessary, see this
<a href="https://blog.opengwas.io/posts/user-auth-spring-2024/">blog
post</a>.</p>
<p>To authenticate, you need to generate a token from the OpenGWAS
website. The token behaves like a password, and it will be used to
authorise the requests you make to the OpenGWAS API. Here are the steps
to generate the token and then have <code>ieugwasr</code> automatically
use it for your queries:</p>
<ol style="list-style-type: decimal">
<li>Login to <a href="https://api.opengwas.io/profile/" class="uri">https://api.opengwas.io/profile/</a></li>
<li>Generate a new token</li>
<li>Add <code>OPENGWAS_JWT=&lt;token&gt;</code> to your
<code>.Renviron</code> file. This file could be either in your home
directory or in the working directory of your R session. You can check
the location of your <code>.Renviron</code> file by running
<code>Sys.getenv(&quot;R_ENVIRON_USER&quot;)</code> in R.</li>
<li>Restart your R session</li>
<li>To check that your token is being recognised, run
<code>ieugwasr::get_opengwas_jwt()</code>. If it returns a long random
string then you are authenticated.</li>
<li>To check that your token is working, run <code>user()</code>. It
will make a request to the API for your user information using your
token. It should return a list with your user information. If it returns
an error, then your token is not working.</li>
</ol>
<p>Now any query to OpenGWAS will automatically include your token to
authorise the request.</p>
<p><strong>IMPORTANT NOTE</strong>: Do not share this token with others
as it is equivalent to a password. If you think your token has been
compromised, you can generate a new one from the OpenGWAS website.</p>
<div id="deprecated-google-authentication" class="section level3">
<h3>Deprecated Google authentication</h3>
<p>Note that previously we used Google OAuth2 for authentication, in
order for users to access private datasets to which they had specific
access. This authentication method is no longer supported, and all users
should now use the JWT token method described above. You can delete the
<code>ieugwasr_oauth2</code> directory as it will no longer be
needed.</p>
</div>
<div id="allowance" class="section level3">
<h3>Allowance</h3>
<p>Due to very high usage, we have had to limit the number of queries
that can be made in a given time period. Every user has an allowance
that is reset periodically, and is used based on the queries that you
submit. If this is posing an issue do get in touch and we can discuss
how to manage this. See here for full details on the allowance system:
<a href="https://api.opengwas.io/api/#allowance" class="uri">https://api.opengwas.io/api/#allowance</a>.</p>
</div>
</div>
<div id="general-api-queries" class="section level2">
<h2>General API queries</h2>
<p>The API has a number of endpoints documented <a href="https://api.opengwas.io/api/">here</a>. A general way to access
them in R is using the <code>api_query</code> function. There are two
types of endpoints - <code>GET</code> and <code>POST</code>.</p>
<ul>
<li><code>GET</code> - you provide a single URL which includes the
endpoint and query. For example, for the <code>association</code>
endpoint you can obtain some rsids in some studies, e.g.
<ul>
<li><code>api_query(&quot;associations/ieu-a-2,ieu-a-7/rs234,rs123&quot;)</code></li>
</ul></li>
<li><code>POST</code> - Here you send a “payload” to the endpoint. So,
the path specifies the endpoint and you add a list of query
specifications. This is useful for long lists of rsids being queried,
for example
<ul>
<li><code>api_query(&quot;associations&quot;, query=list(rsid=c(&quot;rs234&quot;, &quot;rs123&quot;), id=c(&quot;ieu-a-2&quot;, &quot;ieu-a-7&quot;)))</code></li>
</ul></li>
</ul>
<p>The <code>api_query</code> function returns a <code>response</code>
object from the <code>httr</code> package. See below for a list of
functions that make the input and output to <code>api_query</code> more
convenient.</p>
</div>
<div id="get-api-status" class="section level2">
<h2>Get API status</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ieugwasr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">api_status</span>()</span></code></pre></div>
</div>
<div id="get-list-of-all-available-studies" class="section level2">
<h2>Get list of all available studies</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gwasinfo</span>()</span></code></pre></div>
</div>
<div id="get-list-of-a-specific-study" class="section level2">
<h2>Get list of a specific study</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gwasinfo</span>(<span class="st">&quot;ieu-a-2&quot;</span>)</span></code></pre></div>
</div>
<div id="extract-particular-associations-from-particular-studies" class="section level2">
<h2>Extract particular associations from particular studies</h2>
<p>Provide a list of variants to be obtained from a list of studies:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">associations</span>(<span class="at">variants=</span><span class="fu">c</span>(<span class="st">&quot;rs123&quot;</span>, <span class="st">&quot;7:105561135&quot;</span>), <span class="at">id=</span><span class="fu">c</span>(<span class="st">&quot;ieu-a-2&quot;</span>, <span class="st">&quot;ieu-a-7&quot;</span>))</span></code></pre></div>
<p>By default this will look for LD proxies using 1000 genomes reference
data (Europeans only, the reference panel has INDELs removed and only
retains SNPs with MAF &gt; 0.01). This behaviour can be turned off using
<code>proxies=0</code> as an argument.</p>
<p>Note that the queries are performed on rsids, but chromosome:position
values will be automatically converted. A range query can be done using
e.g.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">associations</span>(<span class="at">variants=</span><span class="st">&quot;7:105561135-105563135&quot;</span>, <span class="at">id=</span><span class="fu">c</span>(<span class="st">&quot;ieu-a-2&quot;</span>), <span class="at">proxies=</span><span class="dv">0</span>)</span></code></pre></div>
</div>
<div id="get-the-tophits-from-a-study" class="section level2">
<h2>Get the tophits from a study</h2>
<p>The tophits can be obtained using</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tophits</span>(<span class="at">id=</span><span class="st">&quot;ieu-a-2&quot;</span>)</span></code></pre></div>
<p>Note that it will perform strict clumping by default (r2 = 0.001 and
radius = 10000kb). This can be turned off with <code>clump=0</code>.</p>
</div>
<div id="perform-phewas" class="section level2">
<h2>Perform PheWAS</h2>
<p>Lookup association of specified variants across every study,
returning at a particular threshold. Note that no LD proxy lookups are
made here.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">phewas</span>(<span class="at">variants=</span><span class="st">&quot;rs1205&quot;</span>, <span class="at">pval=</span><span class="fl">1e-5</span>)</span></code></pre></div>
<p>PheWAS can also be performed in only specific subsets of the data.
The datasets in the IGD are organised by batch, you can see info about
it here: <a href="https://gwas.mrcieu.ac.uk/datasets/" class="uri">https://gwas.mrcieu.ac.uk/datasets/</a> or get a list of
batches and their descriptions using:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">batches</span>()</span></code></pre></div>
<p>You can perform PheWAS in only specified batches using:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">phewas</span>(<span class="at">variants=</span><span class="st">&quot;rs1205&quot;</span>, <span class="at">pval=</span><span class="fl">1e-5</span>, <span class="at">batch=</span><span class="fu">c</span>(<span class="st">&#39;ieu-a&#39;</span>, <span class="st">&#39;ukb-b&#39;</span>))</span></code></pre></div>
<p>By default PheWAS is performed in all batches (which is of course
somewhat slower).</p>
</div>
<div id="ld-clumping" class="section level2">
<h2>LD clumping</h2>
<p>The API has a wrapper around <a href="https://www.cog-genomics.org/plink/1.9">plink version 1.90</a> and
can use it to perform clumping with an LD reference panel from 1000
genomes reference data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">tophits</span>(<span class="at">id=</span><span class="st">&quot;ieu-a-2&quot;</span>, <span class="at">clump=</span><span class="dv">0</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">ld_clump</span>(</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">rsid=</span>a<span class="sc">$</span>name, <span class="at">pval=</span>a<span class="sc">$</span>p, <span class="at">id=</span>a<span class="sc">$</span>id)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>There are 5 super-populations that can be requested via the
<code>pop</code> argument. By default this will use the Europeans subset
(EUR super-population). The reference panel has INDELs removed and only
retains SNPs with MAF &gt; 0.01 in the selected population.</p>
<p>Note that you can perform the same operation locally if you provide a
path to plink and a bed/bim/fam LD reference dataset. e.g.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ld_clump</span>(</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">rsid=</span>a<span class="sc">$</span>name, <span class="at">pval=</span>a<span class="sc">$</span>p, <span class="at">id=</span>a<span class="sc">$</span>id),</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">plink_bin =</span> <span class="st">&quot;/path/to/plink&quot;</span>,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">bfile =</span> <span class="st">&quot;/path/to/reference_data&quot;</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>See the following vignette for more information: <a href="local_ld.html">Running local LD operations</a></p>
</div>
<div id="ld-matrix" class="section level2">
<h2>LD matrix</h2>
<p>Similarly, a matrix of LD r values can be generated using</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ld_matrix</span>(b<span class="sc">$</span>variant)</span></code></pre></div>
<p>This uses the API by default but is limited to only 500 variants. You
can use, instead, local plink and LD reference data in the same manner
as in the <code>ld_clump</code> function, e.g.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ld_matrix</span>(b<span class="sc">$</span>variant, <span class="at">plink_bin =</span> <span class="st">&quot;/path/to/plink&quot;</span>, <span class="at">bfile =</span> <span class="st">&quot;/path/to/reference_data&quot;</span>)</span></code></pre></div>
<p>There are 5 super-populations that can be requested via the
<code>pop</code> argument. By default this will use the Europeans subset
(EUR super-population). The reference panel has INDELs removed and only
retains SNPs with MAF &gt; 0.01 in the selected population.</p>
<p>Super-populations:</p>
<ul>
<li>EUR = European</li>
<li>EAS = East Asian</li>
<li>AMR = Admixed American</li>
<li>SAS = South Asian</li>
<li>AFR = African</li>
</ul>
<p>See the following vignette for more information: <a href="local_ld.html">Running local LD operations</a></p>
</div>
<div id="variant-information" class="section level2">
<h2>Variant information</h2>
<p>Translating between rsids and chromosome:position, while also getting
other information, can be achieved.</p>
<p>The <code>chrpos</code> argument can accept the following</p>
<ul>
<li><code>&lt;chr&gt;:&lt;position&gt;</code></li>
<li><code>&lt;chr&gt;:&lt;start&gt;-&lt;end&gt;</code></li>
</ul>
<p>For example</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">variants_chrpos</span>(<span class="fu">c</span>(<span class="st">&quot;7:105561135-105563135&quot;</span>, <span class="st">&quot;10:44865737&quot;</span>))</span></code></pre></div>
<p>This provides a table with dbSNP variant IDs, gene info, and various
other metadata. Similar data can be obtained from searching by rsid</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">variants_rsid</span>(<span class="fu">c</span>(<span class="st">&quot;rs234&quot;</span>, <span class="st">&quot;rs333&quot;</span>))</span></code></pre></div>
<p>And a list of variants within a particular gene region can also be
found. Provide a ensembl or entrez gene ID (e.g. ENSG00000123374 or
1017) to the following:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>c <span class="ot">&lt;-</span> <span class="fu">variants_gene</span>(<span class="st">&quot;ENSG00000123374&quot;</span>)</span></code></pre></div>
</div>
<div id="extracting-gwas-summary-data-based-on-gene-region" class="section level2">
<h2>Extracting GWAS summary data based on gene region</h2>
<p>Here is an example of how to obtain summary data for some datasets
for a gene region. As an example, we’ll extract CDK2 (HGNC number 1017)
from a BMI dataset (ieu-a-2)</p>
<p>Use the <a href="https://bioconductor.org/packages/release/bioc/html/mygene.html">mygene</a>
bioconductor package to query the <a href="https://mygene.info/">mygene.info</a> API.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mygene)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> mygene<span class="sc">::</span><span class="fu">getGene</span>(<span class="st">&quot;1017&quot;</span>, <span class="at">fields=</span><span class="st">&quot;genomic_pos_hg19&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">paste0</span>(a[[<span class="dv">1</span>]]<span class="sc">$</span>genomic_pos_hg19<span class="sc">$</span>chr, <span class="st">&quot;:&quot;</span>, a[[<span class="dv">1</span>]]<span class="sc">$</span>genomic_pos_hg19<span class="sc">$</span>start, <span class="st">&quot;-&quot;</span>, a[[<span class="dv">1</span>]]<span class="sc">$</span>genomic_pos_hg19<span class="sc">$</span>end)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> ieugwasr<span class="sc">::</span><span class="fu">associations</span>(r, <span class="st">&quot;ieu-a-2&quot;</span>)</span></code></pre></div>
</div>
<div id="genomes-annotations" class="section level2">
<h2>1000 genomes annotations</h2>
<p>The OpenGWAS database contains a database of population annotations
from the 1000 genomes project - the alternative allele frequencies and
the LD scores for each variant, calculated for each super population
separately. Only variants are present if they are MAF &gt; 1% in at
least one super population. You can access this info in different
ways</p>
<ol style="list-style-type: decimal">
<li><p>Look up a particular set of rsids</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>ieugwasr<span class="sc">::</span><span class="fu">afl2_rsid</span>(<span class="fu">c</span>(<span class="st">&quot;rs234&quot;</span>, <span class="st">&quot;rs123&quot;</span>))</span></code></pre></div></li>
<li><p>Look up a set of positions or regions</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>ieugwasr<span class="sc">::</span><span class="fu">afl2_chrpos</span>(<span class="st">&quot;1:100000-900000&quot;</span>)</span></code></pre></div></li>
<li><p>Extract annotations for a list of 20k variants that are common in
all super populations, and evenly spaced across the genome</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ieugwasr<span class="sc">::</span><span class="fu">afl2_list</span>()</span></code></pre></div></li>
<li><p>Extract annotations for a 1.3 million HapMap3 variants</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>ieugwasr<span class="sc">::</span><span class="fu">afl2_list</span>(<span class="st">&quot;hapmap3&quot;</span>)</span></code></pre></div></li>
<li><p>Infer the ancestry of a particular study by comparing the allele
frequencies with different super population reference frequencies</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>snplist <span class="ot">&lt;-</span> ieugwasr<span class="sc">::</span><span class="fu">afl2_list</span>()</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>eur_example <span class="ot">&lt;-</span> <span class="fu">associations</span>(snplist<span class="sc">$</span>rsid, <span class="st">&quot;ieu-a-2&quot;</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>ieugwasr<span class="sc">::</span><span class="fu">infer_ancestry</span>(eur_example, snplist)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>eas_example <span class="ot">&lt;-</span> <span class="fu">associations</span>(snplist<span class="sc">$</span>rsid, <span class="st">&quot;bbj-a-10&quot;</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>ieugwasr<span class="sc">::</span><span class="fu">infer_ancestry</span>(eur_example, snplist)</span></code></pre></div></li>
</ol>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
